openedx/features/learner_profile/static/learner_profile/js/learner_profile_factory.js

```diff
@@ -131,7 +131,17 @@
                     options: options.language_options,
                     helpMessage: '',
                     persistChanges: true
+                }),
+
+                new FieldsView.ReadonlyFieldView({
```

Comment:
@robrap it does make sense to me to have a new date field type. Note that it should render a span with the appropriate data attributes that moment.js expects. You then also need to include the following snippet to load the date-utils (which in turn loads moment.js):

https://github.com/edx/edx-platform/blob/5ed622d6d53aa90ead230139b3a9d6d9c97b093e/openedx/features/course_experience/templates/course_experience/course-outline-fragment.html#L158

An example of the span can be seen just above that line:

https://github.com/edx/edx-platform/blob/5ed622d6d53aa90ead230139b3a9d6d9c97b093e/openedx/features/course_experience/templates/course_experience/course-outline-fragment.html#L137

Hopefully this can all be encapsulated into a date field.

Does that help?